minimal_freertos_example/main.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * FreeRTOS Kernel <DEVELOPMENT BRANCH>
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/* This is a simple main  that will start freertos and run a periodic task */
+/* if compiled with the template port, this project will do nothing. */
+
+#include <FreeRTOS.h>
+#include <task.h>
+#include <queue.h>
+#include <timers.h>
+#include <semphr.h>
+
+#include <stdio.h>
+
+StackType_t exampleTaskStack[ configMINIMAL_STACK_SIZE ];
+StaticTask_t exampleTaskTCB;
+
+void exampleTask( void * parameters )
+{
+    for( ; ; )
+    {
+        /* Example Task Code */
+        vTaskDelay( 100 ); /* delay 100 ticks */
+    }
+}
+
+int main( void )
+{
+    printf( "Example FreeRTOS Project\n" );
+
+    xTaskCreateStatic( exampleTask, "example", configMINIMAL_STACK_SIZE, NULL, configMAX_PRIORITIES - 1, exampleTaskStack, &exampleTaskTCB );
+
+    vTaskStartScheduler();
+
+    for( ; ; ) /* should never get here. */
+    {
+    }
+
+    return 0;
+}
+
+void vApplicationStackOverflowHook( TaskHandle_t xTask,
+                                    char * pcTaskName )
+{
+}
+
+void vApplicationGetTimerTaskMemory( StaticTask_t ** ppxTimerTaskTCBBuffer,
+                                     StackType_t ** ppxTimerTaskStackBuffer,
+                                     uint32_t * pulTimerTaskStackSize )
+{
+    static StaticTask_t xTimerTaskTCB;
+    static StackType_t uxTimerTaskStack[ configTIMER_TASK_STACK_DEPTH ];
+
+    *ppxTimerTaskTCBBuffer = &xTimerTaskTCB;
+
+    *ppxTimerTaskStackBuffer = uxTimerTaskStack;
+
+    *pulTimerTaskStackSize = sizeof( uxTimerTaskStack ) / sizeof( *uxTimerTaskStack );
```

Comment:
Wait why is this not just doing what we do in the Posix_GCC demo's [vApplicationGetTimerTaskMemory](https://github.com/FreeRTOS/FreeRTOS/blob/d73470e9367888531f26954b4681de38cd9bcb22/FreeRTOS/Demo/Posix_GCC/main.c#L431)?
`    *pulTimerTaskStackSize = configTIMER_TASK_STACK_DEPTH;`